json11.cpp

```diff
@@ -375,38 +375,38 @@ struct JsonParser final {
       if (str[i] == '/') {
         i++;
         if (i == str.size())
-          return fail("unexpected end of input inside comment", 0);
+          return !!fail("unexpected end of input inside comment", 0);
```

Comment:
The conversion here seems redundant, since this invocation of fail is explicitly requesting a return value of "0" via the final argument (see the template above).  I think you can address this by replacing "0" with "false" instead, and avoid the !!